### DISCUSSION TITLE
Hovercard references to comments in current object should include (relative) dates

### DISCUSSION THREAD

**Discussion by jsoref** [community/community#17020] [Pull Requests]
## Backstory

I'm working on a thing where a bot tries to have a casual conversation with a user in a PR.

I have a sample PR where I've been experimenting with various parts of this adventure...

Here's a starting comment
https://github.com/jsoref/examples-testing/pull/11#issuecomment-1086979293 (it's minimized, at the moment it appears that loading the link in the url will take you to close to the right place, but won't highlight anything, which means you won't know where you're supposed to look -- technically out of scope for this discussion, but i'd love for it to get a blue highlight, I may have already raised that):

![image](https://user-images.githubusercontent.com/2119212/169609510-492042c0-4257-42af-b632-0d8045ebccf3.png)

Click top _This comment has been minimized._ and then you can click:
![image](https://user-images.githubusercontent.com/2119212/169609599-7716ed1d-fd65-4ee5-84d8-8f817b2a037b.png)

this confirms you are where you are.

The comment is _minimized_ (to find out why I _believe_ requires using the GitHub API w/ graphql and probably using a PAT).

Anyway, in the comment, you'll see this:

![image](https://user-images.githubusercontent.com/2119212/169609688-f14add9e-7a04-4689-a9bb-f33233b1b8e6.png)

Which is written as:
```
To have the bot do this for you, reply quoting the following line:
@check-spelling-bot apply [changes](https://github.com/jsoref/examples-testing/pull/11#issuecomment-1086979293).
```

Which is approximately this (if you were to copy and paste it):
> To have the bot do this for you, reply quoting the following line:
> @check-spelling-bot apply https://github.com/jsoref/examples-testing/pull/11#issuecomment-1086979293.

And, as you can see above, copying the text converts the word `changes` into a link which is made pretty by github.

## Hovercards

If you hover over that last link (here) in this discussion (or as I am in this composer box for the discussion), you'll see something like this:

![image](https://user-images.githubusercontent.com/2119212/169609913-a05bbabf-4f0a-4d6f-8f81-aa13c914e278.png)

* The first row says when the PR was opened
* The second row does *not* say when the comment was made -- this is the core to my request
* The third row tells me information about my relation to the PR itself

### Copying that text and commenting

Yields something like:
https://github.com/jsoref/examples-testing/pull/11#issuecomment-1133191817

![image](https://user-images.githubusercontent.com/2119212/169610184-7e0fca59-ddb2-4a90-a431-43513f24deee.png)

And hovering yields the preceding hovercard (or one sufficiently indistinguishable so as to be problematic, i.e. why we're here):
![image](https://user-images.githubusercontent.com/2119212/169610204-55e465d0-a321-4122-8e97-85e57e1a5225.png)

The bot has an opinion which it would like to share:
https://github.com/jsoref/examples-testing/pull/11#issuecomment-1133192237

![image](https://user-images.githubusercontent.com/2119212/169610333-8b5d4351-b400-40fe-b501-b7476f8165f6.png)

Note that both the first and second sentences have links to things.

The first link should look familiar:
https://github.com/jsoref/examples-testing/pull/11#issuecomment-1133191817
since it's a link to a comment from above.

But hovering yields:
![image](https://user-images.githubusercontent.com/2119212/169610433-a8d28dc0-e3ba-43ae-9a36-f3dac1c7bf53.png)

Whereas hovering over the second link yields:
![image](https://user-images.githubusercontent.com/2119212/169610452-92ad88a9-c9a0-4c93-bbd6-926df5cecab7.png)

As you can see, there's an intervening cross link: https://github.com/jsoref/examples-testing/pull/11#ref-issue-1243570386

And then another attempt to talk to the bot https://github.com/jsoref/examples-testing/pull/11#issuecomment-1133316861 

This time, the bot tries to be more helpful in explaining why things aren't working out:
https://github.com/jsoref/examples-testing/pull/11#issuecomment-1133317613
![image](https://user-images.githubusercontent.com/2119212/169610598-e436416a-2f0a-433b-a2ae-976b50578328.png)

Unfortunately, `#11 (comment)` now appears more than once in the comment, and each time it means something different.
![image](https://user-images.githubusercontent.com/2119212/169610658-292685b2-810b-4dba-a38b-2b59ebcc97bc.png)
![image](https://user-images.githubusercontent.com/2119212/169610676-b40f10c8-073f-46b5-aa36-2fc69e66b2f0.png)
But you can't really tell -- again, having a datestamp in that second row would help immensely.

I tried with some variations:
https://github.com/jsoref/examples-testing/pull/11#issuecomment-1133327964

But then I tried to address the duplicated `#11 (comment)` bit:
https://github.com/jsoref/examples-testing/pull/11#issuecomment-1133354074

![image](https://user-images.githubusercontent.com/2119212/169610783-464e128a-2d96-41c1-acc7-d13a66f87164.png)

And now, each hovercard is identical, and even less helpful:
![image](https://user-images.githubusercontent.com/2119212/169610805-8c44168d-81e9-4181-8cbe-a9c49e80876f.png)
![image](https://user-images.githubusercontent.com/2119212/169610820-0071795d-6c5c-4e3f-b0cf-708f15bf89dc.png)
![image](https://user-images.githubusercontent.com/2119212/169610843-f6eb78eb-8644-42e5-8b25-024889568026.png)

---

# TL;DR

1. I'd love for comment references to include datestamps for their comments
2. It'd be nice if self-references to issues/PRs (that included comment-refs) always included the comment portion of hovercards, even if their was a pretty link `[pretty link](#comment-ref)` instead of a `#comment-ref`.
